app/src/main/java/com/weatherxm/ui/home/HomeActivity.kt

```diff
@@ -34,6 +37,14 @@ class HomeActivity : BaseActivity(), BaseMapFragment.OnMapDebugInfoListener {
     private lateinit var binding: ActivityHomeBinding
     private lateinit var navController: NavController
 
+    init {
```

Comment:
Why we are using init & withCreate here and not move the permission's request inside onCreate() ?

Reply:
I thought it was cleaner and also in order to do it in a new Coroutine and not on the main thread (e.g. block the main thread), is it wrong or am I missing something?

Reply:
It's not wrong, I was just curious about the thought behind the decision. 

I could see scope error when using class init, in Activity or Fragment lifecycle but in this specific case, we may not have any. You could also "not block" the main thread, by using another coroutine scope within the onCreate(). I don't have a strong opinion on this topic, so you could try whatever you think is best for this case.